### PR TITLE
tray: when a service is lost, remove all matching items

### LINKED ISSUE #3699: Stale tray icons on program exit
Last checked with `sway version 1.0-rc2`

If one AppIndicator program registers multiple icons, then there is a stale icon in tray on program exit.

Demo program: https://github.com/gnidorah/appindicator-tray-demo

Screenshots:
1. The program is running and registered two icons

2. The program has exited and left one stale icon

cc @ianyfan @4e554c4c @mckinney-subgraph

### PR DESCRIPTION
**Pull request by ianyfan**
Fixes #3699

### CODE REVIEW

**Comment by ianyfan** [comment]
Eh, the condition isn't quite right, don't merge yet.

**Comment by gnidorah** [comment]
@ianyfan Works fine for me. Thank you much!

**Comment by emersion** [comment]
@ianyfan Is this ready for review again?

**Comment by ianyfan** [comment]
Yes, I fixed the check.

File: swaybar/tray/watcher.c [hunk 1]
```diff
@@ -33,18 +29,23 @@ static int handle_lost_service(sd_bus_message *msg,
...
-				using_standard_protocol(watcher) ? cmp_id : cmp_service, service);
-		if (idx != -1) {
+		for (int idx = 0; idx < watcher->items->length; ++idx) {
 			char *id = watcher->items->items[idx];
-			sway_log(SWAY_DEBUG, "Unregistering Status Notifier Item '%s'", id);
-			list_del(watcher->items, idx);
-			sd_bus_emit_signal(watcher->bus, obj_path, watcher->interface,
-					"StatusNotifierItemUnregistered", "s", id);
-			free(id);
+			sway_log(SWAY_DEBUG, "DEBUG %s %s", id, service);
```
Comment (emersion) [inline swaybar/tray/watcher.c]: This could be removed

File: swaybar/tray/watcher.c [hunk 2]
```diff
@@ -33,18 +29,23 @@ static int handle_lost_service(sd_bus_message *msg,
...
+			if ((using_standard_protocol(watcher) ? cmp_id(id, service) :
```
Comment (c-edw) [inline swaybar/tray/watcher.c]: IMO embedding a ternary in a condition is a bit nasty. I would recommend declaring a bool.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**